satpy/composites/spectral.py

```diff
@@ -68,3 +70,54 @@ def __call__(self, projectables, optional_datasets=None, **attrs):
         new_green = sum(fraction * value for fraction, value in zip(self.fractions, projectables))
         new_green.attrs = combine_metadata(*projectables)
         return super(GreenCorrector, self).__call__((new_green,), **attrs)
+
+
+class NDVIHybridGreen(GenericCompositor):
+    """Construct a hybrid green channel weigthed by NDVI.
+
+    This NDVIHybridGreen correction follows the same approach as the HybridGreen compositor, but with a dynamic blend
+    factor `f` that depends on the pixel-level Normalized Differece Vegetation Index (NDVI). The higher the NDVI, the
+    smaller the contribution from the nir channel will be, following a liner relationship between the two ranges
+    [ndvi_min, ndvi_max] and `fractions`.
+
+    A new green channel using e.g. FCI data and the NDVIHybridGreen compositor can be defined like:
+
+    ndvi_hybrid_green:
+        compositor: !!python/name:satpy.composites.spectral.NDVIHybridGreen
+        fractions: [0.15, 0.05]
+        prerequisites:
+          - name: vis_05
+            modifiers: [sunz_corrected, rayleigh_corrected]
+          - name: vis_08
+            modifiers: [sunz_corrected, rayleigh_corrected]
+        standard_name: toa_bidirectional_reflectance
+
+    In this example, pixels with NDVI=0.0 (default `ndvi_min`) will be a weighted average with 85% contribution from the
```

Comment:
Well this is the interesting bit with the green band correction. The location of the AHI/FCI green bands has two effects:
1. Vegetation appears brownish.
2. Barren appears reddish.

The green color of vegetation can be boosted with a higher contribution from the NIR channel. Similarly, the reddishness of barren is reduced with an increasing contribution from the NIR channel. However, more contribution from NIR is required to get the proper color of barren, compared to vegetation. Hence, I came up with the solution of increasing the contribution from NIR with decreasing NDVI.

Below are a few examples using the default hybrid green method, with decreasing contribution from the NIR channel. As you can see the color of Australia starts to get too red when the color of vegetation start to look realistic. I also included the result of the NDVIHybridGreen (with just qualitative tuning) and VIIRS RGB from NASA worldview as reference.

| 15% NIR | 10% NIR | 7% NIR | 4% NIR | 0% NIR | With this PR | VIIRS from NASA Worldview |
| --- | --- | --- | --- | --- | --- | --- |
| ![AHI_true_color_hybrid_green_85_15](https://user-images.githubusercontent.com/42137969/202488382-26de85f4-db0b-4277-afc6-fde4d3f7155b.png) | ![AHI_true_color_hybrid_green_90_10](https://user-images.githubusercontent.com/42137969/202488429-b9d740d0-88d4-4e87-8773-e2c3a2ff0187.png) | ![AHI_true_color_hybrid_green_93_7](https://user-images.githubusercontent.com/42137969/202488450-cf69ac84-b266-471a-b9dc-a3bf10780a4f.png) | ![AHI_true_color_hybrid_green_96_4](https://user-images.githubusercontent.com/42137969/202488500-b2ae48b1-6657-483d-ada5-cc16be4afd65.png) | ![AHI_true_color_hybrid_green_100_0](https://user-images.githubusercontent.com/42137969/202488658-88ac5469-96da-49cc-93ff-99347a03740d.png) | ![AHI_true_color_ndvi_green_15_5](https://user-images.githubusercontent.com/42137969/202488850-2b452147-ceb7-4105-8636-b3902b208353.png) | ![image](https://user-images.githubusercontent.com/42137969/202488902-994db9ba-82c6-498f-b36f-c65b080605c4.png) |

Reply:
This is only internal developments in preparation for FCI. We are testing with AHI since we have the same scenario with sub-optimal position of the green band there. Hopefully we can publish it at some point though.